### PR TITLE
Task4 s1 pop database

### PR DESCRIPTION
**Pull request by jaholt**

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -17,10 +18,66 @@ function grabApi () {
...
+
+/*
+ * inserts our Pokemon data in the database
+ * $finalValues - our concacinated arrays with an trim ob the final Pokemon to remove the final ','.
+ * $db - PDO to our database;
+ */
+
+function database ($finalValues, $db){
+
+    $query = $db->prepare("INSERT INTO `pokemon` (`pokemon_Name`, `pokemon_type`, `pokemon_type2`) VALUES " . $finalValues .";");
```
Comment (mattap86) [inline index.php]: query needs refactoring: pokemon_Name to pokemon_name; pokemon_type2 to pokemon_type_2

File: index.php [hunk 2]
```diff
@@ -17,10 +18,66 @@ function grabApi () {
...
+
+    return $query->execute();
+
```
Comment (mattap86) [inline index.php]: unnecessary lb

File: index.php [hunk 3]
```diff
@@ -17,10 +18,66 @@ function grabApi () {
...
+}
+
+$pokemonArray = grabApi();
+
+$db = new PDO('mysql:host=127.0.0.1; dbname=Pokedex', 'root');
+
+//$allPokemon = getPokeType($pokemonArray);
```
Comment (mattap86) [inline index.php]: does this need uncommenting or removing?

File: index.php [hunk 4]
```diff
@@ -17,10 +18,66 @@ function grabApi () {
...
-grabApi();
+/*
+ * Allows us to insert all of our data in our DB at once. By putting each into a () we can concatinate each and then insert into the db as one large value.
+ *
+ *$allPokemen - the array of all the pokemon (with the only data we needed).
+ *
+ * $values - stores the concacinated arrays.
+ * $finalValues - our concacinated arrays with an trim ob the final Pokemon to remove the final ','.
+ */
+
```
Comment (mattap86) [inline index.php]: unnecessary lb

File: index.php [hunk 5]
```diff
@@ -17,10 +18,66 @@ function grabApi () {
...
     }
     curl_close($curl);
     return $pokemon;
 }
 
-grabApi();
+/*
+ * Allows us to insert all of our data in our DB at once. By putting each into a () we can concatinate each and then insert into the db as one large value.
+ *
+ *$allPokemen - the array of all the pokemon (with the only data we needed).
```
Comment (mattap86) [inline index.php]: docblock need srewording: @param for params. @return for result with datatype

File: index.php [hunk 6]
```diff
@@ -17,10 +18,66 @@ function grabApi () {
...
+ *
+ * $values - stores the concacinated arrays.
+ * $finalValues - our concacinated arrays with an trim ob the final Pokemon to remove the final ','.
```
Comment (mattap86) [inline index.php]: typo: concacinated to concatenated.

File: index.php [hunk 7]
```diff
@@ -17,10 +18,66 @@ function grabApi () {
...
+ */
+
+function insertintodatabase($allPokemon) {
```
Comment (mattap86) [inline index.php]: type hint param and type hint return?

File: index.php [hunk 8]
```diff
@@ -17,10 +18,66 @@ function grabApi () {
...
+    foreach ($allPokemon as $result){
+        $values .= '(\'' . $result[0] . '\',';
+        $values .= '\'' . $result[1] . '\',';
+       if (isset($result[2])) {
+           $values .= '\'' . $result[2] . '\'), ';
+       } else {
+           $values .= '\'' . 'NULL' . '\'), ';
+       }
+    }
+
```
Comment (olivergate) [inline index.php]: unnecessary  LB

File: index.php [hunk 9]
```diff
@@ -2,10 +2,11 @@
...
 /*
  * This is an API call to the pokeAPI and pulls all of the data from 151 pokemon and puts them in an array. thew sleep stops the api call half way through for 60 seconds
+ *$Temp stores each Pokemon data for a limited time, allowing for the needed data to be removed and stored in pokemon before being discarded on the next loop.
  *
- *
- * @result this is an array with a json string in it containing of all of the information of all of the 151 pokemon called
+ * $pokemon - holds all of our pokemon, with the only data we need from each, name, type 1 and type 2.
  */
+
 function grabApi () {
```
Comment (mattap86) [inline index.php]: type hint return?

File: index.php [hunk 10]
```diff
@@ -17,10 +18,66 @@ function grabApi () {
         }
         $i ++;
         curl_setopt($curl, CURLOPT_URL, 'https://pokeapi.co/api/v2/pokemon/' . $i . '/');
-        $pokemon[] = curl_exec($curl);
+        $resp = curl_exec($curl);
+        $temp = json_decode($resp, TRUE);
+        $name = $temp['forms'][0]['name'];
+        $type1 = $temp['types'][0]['type']['name'];
+        $type2 = $temp['types'][1]['type']['name'];
```
Comment (mattap86) [inline index.php]: if isset on type 2 to remove notice messages if no type 2 present

File: index.php [hunk 11]
```diff
@@ -17,10 +18,66 @@ function grabApi () {
...
+}
+
+
+/*
+ * inserts our Pokemon data in the database
+ * $finalValues - our concacinated arrays with an trim ob the final Pokemon to remove the final ','.
+ * $db - PDO to our database;
+ */
+
+function database ($finalValues, $db){
```
Comment (mattap86) [inline index.php]: type hinting and type hint return?

File: index.php [hunk 12]
```diff
@@ -2,10 +2,11 @@
 
 /*
  * This is an API call to the pokeAPI and pulls all of the data from 151 pokemon and puts them in an array. thew sleep stops the api call half way through for 60 seconds
+ *$Temp stores each Pokemon data for a limited time, allowing for the needed data to be removed and stored in pokemon before being discarded on the next loop.
  *
- *
- * @result this is an array with a json string in it containing of all of the information of all of the 151 pokemon called
+ * $pokemon - holds all of our pokemon, with the only data we need from each, name, type 1 and type 2.
```
Comment (olivergate) [inline index.php]: DOC BLOC is not formatted to PSR2 @result necessary

File: index.php (hunk 5 above)
Comment (olivergate) [inline index.php]: DOC BLOC  nedds to be written according to PSR2 standards. @param and @results missing.

File: index.php (hunk 9 above)
Comment (olivergate) [inline index.php]: Return hinting required

File: index.php (hunk 7 above)
Comment (olivergate) [inline index.php]: parameter and return hinting necessary.

File: index.php [hunk 13]
```diff
@@ -17,10 +18,66 @@ function grabApi () {
...
+ * $finalValues - our concacinated arrays with an trim ob the final Pokemon to remove the final ','.
+ */
+
+function insertintodatabase($allPokemon) {
+    $values = '';
+
+    foreach ($allPokemon as $result){
+        $values .= '(\'' . $result[0] . '\',';
+        $values .= '\'' . $result[1] . '\',';
+       if (isset($result[2])) {
```
Comment (olivergate) [inline index.php]: Incorrect indentation

File: index.php [hunk 14]
```diff
@@ -17,10 +18,66 @@ function grabApi () {
...
+    }
+
+
+    $finalValues = rtrim($values, ', ');
+    return $finalValues;
+}
+
+
+/*
+ * inserts our Pokemon data in the database
```
Comment (olivergate) [inline index.php]: INcorrect DOC BLOC Style. PSR2 requires the function description at the top with a space underneath and then @param and @return are required (with hinting)

File: index.php (hunk 11 above)
Comment (olivergate) [inline index.php]: type hinting and return hinting missing.

File: index.php (hunk 3 above)
Comment (olivergate) [inline index.php]: Is this necessary?

**Pull request merged**